### PR TITLE
Adding gitlab CI base config

### PR DESCRIPTION
**Pull request by davidferlay**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    --standard=Drupal,DrupalPractice
+    --extensions=php,module,inc,install,profile,theme,yml,txt,md,js
+    --ignore=*.css,libraries/*
+    web/profiles
+    web/modules/custom
+    web/themes/custom
+
+sniffers:sonarqube:
+  stage: sniffers
+  image: newtmitch/sonar-scanner:3
```
Comment (andypost) [inline .gitlab-ci.yml]: please use `skilldlabs/sonar-scanner:3.3`

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+stop_review:
+  <<: *job_deploy_commit
+  variables:
+    GIT_STRATEGY: none
+  when: manual
+  environment:
+    name: review/$CI_COMMIT_REF_NAME
+    action: stop
+
+backup:git:
```
Comment (andypost) [inline .gitlab-ci.yml]: not sure it needed

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+variables:
+  GIT_DEPTH: "3"
+
+before_script:
+  - date
+  - pwd
+
+stages:
+  - sniffers
```
Comment (andypost) [inline .gitlab-ci.yml]: please add `sniffers:composer` with basic `time composer validate` script

File: .gitlab-ci.yml (hunk 1 above)
Comment (davidferlay) [reply .gitlab-ci.yml]: Done

File: .gitlab-ci.yml (hunk 2 above)
Comment (davidferlay) [reply .gitlab-ci.yml]: Agreed, removing

File: .gitlab-ci.yml (hunk 3 above)
Comment (davidferlay) [reply .gitlab-ci.yml]: Job was added)

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  - backup
```
Comment (andypost) [inline .gitlab-ci.yml]: this stage should be removed

Comment (davidferlay) [reply .gitlab-ci.yml]: I had missed it, now removed

**Pull request merged**